client/client.go

```diff
@@ -81,6 +85,17 @@ type Client struct {
 	manualOverride bool
 }
 
+func redirectError(req *http.Request, via []*http.Request) error {
+	// From Go 1.8 the HTTP client redirect behavior is changed:
+	// HTTP respsonse with 301, 307, and 308 will be automatically redirected.
+	// This behavior change manifests in the client in that 301 is now followed and the client generated an error.
+	// To fix this, `redirectError` is used to maintain a similar behavior to the old one.
```

Comment:
Instead of "similiar behaviour to old one" could you say what the behaviour is?

A short summary of the problem from #29602 would be great. Something like "Previously these error codes would not perform a second request"

Also could you move this above the function to be a godoc string?